src/Module/RenderCartBookingInfoHandler.php

```diff
@@ -191,41 +203,132 @@ public function __invoke()
+        $format   = $this->_containerGet($this->cartItemConfig, 'booking_datetime_format');
+        $clientTz = $this->_getDisplayTimezone($booking);
 
-        try {
-            $clientTz = $this->_containerGet($this->_normalizeContainer($booking), 'client_tz');
-        } catch (InvalidArgumentException $invalidArgumentException) {
-            $clientTz = 'UTC';
-        } catch (NotFoundExceptionInterface $notFoundException) {
-            $clientTz = 'UTC';
-        }
+        // Get timestamps from booking
+        $startTs = $booking->getStart();
+        $endTs   = $booking->getEnd();
 
-        try {
-            $tz = new DateTimeZone($clientTz);
-        } catch (Exception $exception) {
-            $tz = new DateTimeZone('UTC');
+        // Create date time helper instances
+        $startDt = Carbon::createFromTimestampUTC($startTs);
+        $endDt   = Carbon::createFromTimestampUTC($endTs);
+
+        // Shift to client timezone, if available
+        if ($clientTz !== null) {
+            $startDt->setTimezone($clientTz);
+            $endDt->setTimezone($clientTz);
         }
 
-        $startTs  = $booking->getStart();
-        $startDt  = Carbon::createFromTimestampUTC($startTs);
-        $startDt->setTimezone($tz);
-        $startUtc = $startDt->getTimestamp();
+        // Format times to strings
         $startStr = $startDt->format($format);
-
-        $endTs  = $booking->getEnd();
-        $endDt  = Carbon::createFromTimestampUTC($endTs);
-        $endDt->setTimezone($tz);
-        $endUtc = $endDt->getTimestamp();
-        $endStr = $endDt->format($format);
+        $endStr   = $endDt->format($format);
 
         return $this->_getTemplate()->render([
             'from_label'     => $this->__('From:'),
             'until_label'    => $this->__('Until:'),
-            'start_datetime' => $startUtc,
-            'end_datetime'   => $endUtc,
+            'timezone_label' => $this->__('Timezone:'),
+            'start_datetime' => $startTs,
+            'end_datetime'   => $endTs,
             'start_text'     => $startStr,
             'end_text'       => $endStr,
+            'timezone_text'  => $clientTz->getName()
         ]);
     }
+
+    /**
+     * Retrieves the timezone to use for displaying booking dates and times.
+     *
+     * @since [*next-version*]
+     *
+     * @param BookingInterface $booking The booking instance.
+     *
+     * @return DateTimeZone The timezone instance.
+     */
+    protected function _getDisplayTimezone(BookingInterface $booking)
+    {
+        $try = [
+            [$this, '_getBookingClientTimezone'],
+            [$this, '_getFallbackTimezone'],
+            [$this, '_getWordPressTimezone'],
+        ];
+
+        foreach ($try as $_callable) {
+            try {
+                return call_user_func_array($_callable, [$booking]);
+            } catch (Exception $exception) {
+                continue;
```

Comment:
I don't really like the silent behaviour of this, because we can't tell *why* the timezone could not be retrieved, and that reason may be something we need to know about. But will do for now.